### PR TITLE
Badge: Fix the UI mismatch, add a new variant with tests

### LINKED ISSUE #192: Badge: Fix UI Mismatches
GDS: https://share.goabstract.com/cea1e031-2fd9-4fc3-a1f8-407c761b05ef?sha=

### PR DESCRIPTION
**Pull request by ritikamotwani**
Fixes: #192 
To check the changes: https://docs.google.com/document/d/1pHImZ2YXcG0WxMU5ll1y9zBygyErR2EjR8xu3Y2lgDw/edit

### CODE REVIEW

**Comment by tingtingdisco** [comment]
@ritikamotwani could you confirm with designers that do we still need `sup`? thank u

File: src/General/Badge/Badge.tsx [hunk 1]
```diff
@@ -26,6 +33,7 @@ const Badge: React.FunctionComponent<Props> = ({
 interface Props extends React.ComponentPropsWithoutRef<typeof BadgeContainer> {
   label: string | number;
   sup?: boolean;
+  type?: string;
```
Comment (tingtingdisco) [inline src/General/Badge/Badge.tsx]: should be `variant`?

**Comment by tingtingdisco** [comment]
please update storybook page, to include the default type, and update the margin of the examples
Screen Shot 2020-05-25 at 10 58 08 AM
Screen Shot 2020-05-25 at 10 58 39 AM

File: src/General/Badge/Badge.tsx (hunk 1 above)
Comment (ritikamotwani) [reply src/General/Badge/Badge.tsx]: Yes yes, forgot to commit this file after changing the prop names to variant, from type.

**Comment by ritikamotwani** [comment]
> please update storybook page, to include the default type, and update the margin of the examples

Updated the storybook with all default types, and added another div to show the accompanied margin.

**Review by tingtingdisco — APPROVED** [review]
LGTM

**Pull request merged**